src/elements/Asset.php

```diff
@@ -265,6 +266,9 @@ protected static function defineActions(string $source = null): array
                 ]
             );
 
+            // Copy URL
+            $actions[] = CopyAssetUrl::class;
```

Comment:
Wouldn't it be better to do: 
`
if ($volume->hasUrls) {
    $actions[] = CopyAssetUrl::class;
}
`
This would make sure if `hasUrls` is disabled on the volume you don't see the unneeded `Copy URL` button.

Reply:
Ah yes, I somehow missed that `$volume` is available in the static method, thanks!